tyk-pro/scripts/bootstrap_k8s.sh

```diff
@@ -1,28 +1,176 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 set -e
 
-apk --no-cache add curl
+apk --no-cache add curl jq
 curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
 install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
 
-NAMESPACE=tyk
+NAMESPACE=$TYK_POD_NAMESPACE
 
-kubectl exec -it -n ${NAMESPACE} svc/dashboard-svc-tyk-pro -- ./tyk-analytics bootstrap --conf=/etc/tyk-dashboard/tyk_analytics.conf > bootstrapped
+DASHBOARD_HOSTNAME="$TYK_DASHBOARD_SVC.$TYK_POD_NAMESPACE.svc.cluster.local:$TYK_DB_LISTENPORT"
 
-TYK_AUTH=$(awk -F ':' '/USER AUTHENTICATION CODE: /{ print $2 }' bootstrapped | tr -d '[:space:]')
-TYK_ORG=$(awk -F ':' '/ORG ID: /{ print $2 }' bootstrapped | tr -d '[:space:]')
-TYK_MODE=pro
-TYK_URL=http://dashboard-svc-tyk-pro.tyk.svc.cluster.local:3000
-TYK_USER=$(awk -F ':' '/User: /{ print $2 }' bootstrapped | tr -d '[:space:]')
-TYK_PASS=$(awk -F ':' '/Pass: /{ print $2 }' bootstrapped | tr -d '[:space:]')
+main(){
+  if [[ $# -lt 1 ]]
+    then
+      syntax_message
+  elif [ -z $(which curl) ]
+    then
+      curl_message
+    else
+      generate_credentials "$DASHBOARD_HOSTNAME"
+      exit_reason "$?"
+  fi
+}
+
+syntax_message() {
+  echo 'SYNTAX ERROR: Usage `./bootstrap.sh DASHBOARD_HOSTNAME`'
+  exit 1
+}
+
+curl_message() {
+  echo 'ERROR: Please install cURL'
+  exit 1
+}
+
+generate_credentials(){
+  error=0
+  [[ -z "$1" ]] && return "6" || echo ""
+  error="$?"
+  [[ $error != 0 ]] && return $error
+  echo "Creating Organisation"
+  ORGDATA=$(create_organisation)
+  error="$?"
+  [[ $error != 0 ]] && return $error || echo "ORG DATA: $ORGDATA"
+  export ORGID=$(org_id "$ORGDATA")
+  error="$?"
+  [[ $? != 0 ]] && return $error || echo "ORG ID: $ORGID"
+  echo ""
+  echo "Adding new user"
+  export USER_AUTH_CODE=$(create_user $ORGID)
+  error=$?
+}
+
+exit_reason(){
+  case "$1" in
+    "3")
+      echo "ERROR: Unable to connect to the Dashboard. Please check that you are using the correct hostname."
+      exit $1
+      ;;
+    "4")
+      echo "ERROR: Unable to parse JSON"
+      exit $1
+      ;;
+    "5")
+      echo "Unexpected error"
+      exit $1
+      ;;
+    "6")
+      echo "ERROR: Hostname should not contain port number or protocol"
+      ;;
+    *)
+      echo ""
+      ;;
+  esac
+}
+
+if_present_echo(){
+  [[ -z "$1" ]] && return $2 || echo "$1"
+}
+
+create_organisation() {
+  ORGDATA=$(curl --silent --header "admin-auth: $TYK_ADMIN_SECRET" --header "Content-Type:application/json" --data '{"owner_name": "'"$TYK_ORG_NAME"'", "cname_enabled": true, "cname": "'$TYK_ORG_CNAME'"}' $DASHBOARD_HOSTNAME/admin/organisations 2>&1)
```

Comment:
`DASHBOARD_HOSTNAME` variable doesn't appear to include the protocol as seen above:
```
DASHBOARD_HOSTNAME="$TYK_DASHBOARD_SVC.$TYK_POD_NAMESPACE.svc.cluster.local:$TYK_DB_LISTENPORT"
```

Does it work correctly? Same question on the other usages.

Reply:
yes, it works without adding `http://`. I'll add it to the var anyway